app/src/test/java/com/hedvig/app/feature/loggedin/service/TabNotificationServiceTest.kt

```diff
@@ -1,57 +1,50 @@
 package com.hedvig.app.feature.loggedin.service
 
-import androidx.datastore.core.DataStore
-import androidx.datastore.preferences.core.Preferences
 import assertk.assertThat
 import assertk.assertions.isEqualTo
 import com.hedvig.android.owldroid.type.TypeOfContract
 import com.hedvig.app.feature.loggedin.ui.LoggedInTabs
+import com.hedvig.app.service.badge.CrossSellNotificationBadgeService
 import io.mockk.coEvery
-import io.mockk.every
 import io.mockk.mockk
 import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.flow.flowOf
 import kotlinx.coroutines.test.runBlockingTest
 import org.junit.Test
 
 class TabNotificationServiceTest {
+
     @Test
     fun `when all cross-sells have been seen, should not show notification for insurance tab`() {
-        val mockUseCase = mockk<GetCrossSellsUseCase>()
-        coEvery { mockUseCase.invoke() } returns setOf(TypeOfContract.SE_ACCIDENT.toString())
-        val mockDataStore = mockedDataStore(setOf(TypeOfContract.SE_ACCIDENT.toString()))
-
-        val sut = TabNotificationService(
-            mockUseCase,
-            mockDataStore,
+        val mockService = mockedCrossSellNotificationBadgeService(
+            returnValue = emptySet()
         )
 
+        val sut = TabNotificationService(mockService)
+
         runBlockingTest {
-            assertThat(sut.load().first()).isEqualTo(emptySet())
+            assertThat(sut.unseenTabNotifications().first()).isEqualTo(emptySet())
         }
     }
 
     @Test
     fun `when there is an unseen cross-sell, should show notification for insurance tab`() {
-        val mockUseCase = mockk<GetCrossSellsUseCase>()
-        coEvery { mockUseCase.invoke() } returns setOf(TypeOfContract.SE_ACCIDENT.toString())
-        val mockDataStore = mockedDataStore(emptySet())
-
-        val sut = TabNotificationService(
-            mockUseCase,
-            mockDataStore,
+        val mockService = mockedCrossSellNotificationBadgeService(
+            returnValue = setOf(TypeOfContract.SE_ACCIDENT)
         )
 
+        val sut = TabNotificationService(mockService)
+
         runBlockingTest {
-            assertThat(sut.load().first()).isEqualTo(setOf(LoggedInTabs.INSURANCE))
+            assertThat(sut.unseenTabNotifications().first()).isEqualTo(setOf(LoggedInTabs.INSURANCE))
         }
     }
 
-    private fun mockedDataStore(data: Set<String>): DataStore<Preferences> {
-        val mockPreferences = mockk<Preferences>()
-        every { mockPreferences[TabNotificationService.SEEN_CROSS_SELLS_KEY] } returns data
-        val mockDataStore = mockk<DataStore<Preferences>>()
-        every { mockDataStore.data } returns flowOf(mockPreferences)
-        return mockDataStore
+    private fun mockedCrossSellNotificationBadgeService(
+        returnValue: Set<TypeOfContract>
+    ): CrossSellNotificationBadgeService {
+        val mock = mockk<CrossSellNotificationBadgeService>()
```

Comment:
Generally, try to avoid mocking types we own - this often leads to overmocking, resulting in unreliable and hard-to-maintain tests.

A good rule is that things that directly does I/O or interacts with platform code should be mocked 🙂